subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestTaskIntegrationTest.groovy

```diff
@@ -231,6 +231,28 @@ class TestTaskIntegrationTest extends AbstractIntegrationSpec {
         result.assertOutputContains("The setTestClassesDir(File) method has been deprecated and is scheduled to be removed in Gradle 5.0. Please use the setTestClassesDirs(FileCollection) method instead.")
     }
 
+    @Issue("https://github.com/gradle/gradle/issues/3627")
+    def "can reference properties from TestTaskReports when using @CompileStatic"() {
+        buildFile << """
+            import groovy.transform.CompileStatic
+
+            @CompileStatic
+            class StaticallyCompiledPlugin implements Plugin<Project> {
+                @Override
+                void apply(Project project) {
+                    project.apply plugin: 'java'
+                    Test test = (Test) project.tasks.getByName("test")
+                    println test.reports.junitXml.destination
```

Comment:
For good measure should assert the proper destination here or in the `expect` below.

Reply:
I changed the code so we don't need to assert the value since it's not required for the test.